monads/src/FailCont/Main.hs

```diff
@@ -10,10 +10,25 @@ data Error
   deriving (Show, Eq)
 
 addInts :: String -> String -> FailCont r Error Int
-addInts = undefined 
+addInts x y = case (readMaybe x, readMaybe y) of
+  (Nothing, _)       -> FailCont $ \_ fail -> fail EmptyInput
+  (_, Nothing)       -> FailCont $ \_ fail -> fail (ParseFailed y)
+  (Just a, Just b)   -> FailCont $ \success _ -> success (a + b)
```

Comment:
`addInts "a" "42"` results in the `EmptyInput` error, which isn't correct. Similar issue in the second line.